lib/src/frontend/future_matchers.dart

```diff
@@ -80,3 +80,54 @@ class _Completes extends AsyncMatcher {
     return description;
   }
 }
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after 20  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
+/// [timesToPump] passes through the event queue.
+final Matcher doesNotComplete = const _DoesNotComplete(20);
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after [timesToPump]  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
+/// [timesToPump] passes through the event queue.
+Matcher doesNotCompleteAfter(int timesToPump) =>
+    new _DoesNotComplete(timesToPump);
+
+class _DoesNotComplete extends AsyncMatcher {
+  final int timesToPump;
+  const _DoesNotComplete(this.timesToPump);
+
+  Future _pumpEventQueue(times) {
+    if (times == 0) return new Future.value();
+    return new Future(() => _pumpEventQueue(times - 1));
+  }
+
+  /*FutureOr<String>*/ matchAsync(item) {
+    if (item is! Future) return null; // non-Future instances cannot 'complete'
```

Comment:
I think non-futures should be an error here—it's probably a mistake if you pass one in.